### PR TITLE
20155-Browse-implementers-of-in-Nautilus-doesnt-parse-source-code-properly-for-methods-in-a-block

### PR DESCRIPTION
**Pull request by apblack**
Fixes 20155 using a modificaton of teh code in the Bug report.  The second commit removes what appears to be dead code; if this one casues trouble, it oculd be dropped.  (Since the removed methods send unimplemented messages, they are unlikely to help!)

### CODE REVIEW

**Comment by tesonep** [comment]
This PR fixes two issues

https://pharo.fogbugz.com/f/cases/20605
https://pharo.fogbugz.com/f/cases/20155

**Comment by guillep** [comment]
Unrelated tests

File: src/Rubric.package/RubSmalltalkEditor.class/instance/referencesToIt.st [hunk 1]
```diff
@@ -2,9 +2,8 @@ menu messages
...
 	"Open a references browser on the selected symbol"
 
-	| aSymbol env ref |
+	| aSymbol |
 	self selectLine.
-	((aSymbol := self selectedSymbol) == nil)
-			ifTrue: [^ textArea flash].
-
+	(aSymbol := self selectedSymbol) isNil
+		ifTrue: [ ^ textArea flash ].
```
Comment (nicolas-cellier-aka-nice) [inline src/Rubric.package/RubSmalltalkEditor.class/instance/referencesToIt.st]: ifNil: ?

File: src/Rubric.package/RubSmalltalkEditor.class/instance/internalCallToImplementorsOfIt.st [hunk 2]
```diff
@@ -3,8 +3,7 @@ internalCallToImplementorsOfIt
...
 
 	| aSelector |
-	self lineSelectAndEmptyCheck: [^ self].
-	(aSelector := self selectedSelector) == nil ifTrue: [^ textArea flash].
-	self internalCallToImplementorsOf: aSelector.
-			
-	
\ No newline at end of file
+	self lineSelectAndEmptyCheck: [ ^ self ].
+	(aSelector := self selectedSelector) isNil
```
Comment (nicolas-cellier-aka-nice) [inline src/Rubric.package/RubSmalltalkEditor.class/instance/internalCallToImplementorsOfIt.st]: ifNil: ?

**Pull request merged**